user-api/src/main/java/com/example/auction/user/api/UserService.java

```diff
@@ -1,31 +1,37 @@
 package com.example.auction.user.api;
 
 import akka.NotUsed;
+import com.example.auction.pagination.PaginatedSequence;
+import com.example.auction.security.SecurityHeaderFilter;
 import com.lightbend.lagom.javadsl.api.Descriptor;
 import com.lightbend.lagom.javadsl.api.Service;
 import com.lightbend.lagom.javadsl.api.ServiceCall;
 import com.lightbend.lagom.javadsl.api.deser.PathParamSerializers;
-import org.pcollections.PSequence;
 
+import java.util.Optional;
 import java.util.UUID;
 
-import static com.lightbend.lagom.javadsl.api.Service.*;
+import static com.lightbend.lagom.javadsl.api.Service.named;
+import static com.lightbend.lagom.javadsl.api.Service.pathCall;
 
 public interface UserService extends Service {
 
     ServiceCall<UserRegistration, User> createUser();
 
     ServiceCall<NotUsed, User> getUser(UUID userId);
 
-    // Remove once we have a proper user service
-    ServiceCall<NotUsed, PSequence<User>> getUsers();
+    ServiceCall<NotUsed, PaginatedSequence<User>> getUsers(
+            Optional<Integer> pageNo, Optional<Integer> pageSize);
 
     @Override
     default Descriptor descriptor() {
         return named("user").withCalls(
                 pathCall("/api/user", this::createUser),
                 pathCall("/api/user/:id", this::getUser),
-                pathCall("/api/user", this::getUsers)
-        ).withPathParamSerializer(UUID.class, PathParamSerializers.required("UUID", UUID::fromString, UUID::toString));
+                pathCall("/api/user?pageNo&pageSize", this::getUsers)
+        ).withPathParamSerializer(
+                UUID.class, PathParamSerializers.required("UUID", UUID::fromString, UUID::toString)
+        ).withHeaderFilter(SecurityHeaderFilter.INSTANCE);
```

Comment:
What's the reason for adding the `SecurityHeaderFilter`? I don't see it used anywhere.